### DISCUSSION TITLE
transparent material rendering

### DISCUSSION THREAD

**Discussion by tom-deadline** [google/model-viewer#2260] [Q&A]
Hey, guys!

I'm rendering a model of glasses and the transparent material of lenses is rendered with little fragments inside, see the picture below. 
![image (1)](https://user-images.githubusercontent.com/44176198/113708519-5f4bfe80-96f2-11eb-948e-1f88078a5751.png)
Is there a way to fix it?

**Comment by elalish** [comment]
First, ensure your triangles all have consistent winding and don't overlap. However, this is probably just a triangle sorting problem (GPUs do things in a non-deterministic order). I'm guessing you only see this edge on, but not from the front? We do extra work to make sure the triangle sorting is correct for convex objects, but it's not very possible for concave ones. You might consider making the lenses a single surface of double-sided material so that it's not looking through multiple surfaces.

**Reply by elalish** [reply]
You'll also want to ensure you separate transparent objects into separate meshes so they can be rendered in the proper order. Sometimes dividing them even further to help them be more convex is also helpful.
